RuntimeClass GA (#24698)
kubernetes/website: content/en/docs/reference/access-authn-authz/admission-controllers.md

### RuntimeClass {#runtimeclass}

{{< feature-state for_k8s_version="v1.20" state="stable" >}}

If you enable the `PodOverhead` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/), and define a RuntimeClass with [Pod overhead](/docs/concepts/scheduling-eviction/pod-overhead/) configured, this admission controller checks incoming
Pods. When enabled, this admission controller rejects any Pod create requests that have the overhead already set.
For Pods that have a  RuntimeClass is configured and selected in their `.spec`, this admission controller sets `.spec.overhead` in the Pod based on the value defined in the corresponding RuntimeClass.

> {{< note >}}
> The `.spec.overhead` field for Pod and the `.overhead` field for RuntimeClass are both in beta. If you do not enable the `PodOverhead` feature gate, all Pods are treated as if `.spec.overhead` is unset.
> {{< /note >}}

Author: Should I do this? Or it will be too much?

```suggestion
{{< note >}}

{{< feature-state for_k8s_version="v1.18" state="beta" >}}

The `.spec.overhead` field for Pod and the `.overhead` field for RuntimeClass are both in beta. If you do not enable the `PodOverhead` feature gate, all Pods are treated as if `.spec.overhead` is unset.
{{< /note >}}
```

Reviewer: looks like an overkill to me. RuntimeClass is GA but the `.overhead` field in it is beta. The `feature-state` line on 736 may cause some confusion. The text is already pretty clear.

### RuntimeClass {#runtimeclass}

> {{< feature-state for_k8s_version="v1.18" state="beta" >}}

Reviewer 2: Shouldn't this be v1.20 stable? The admission controller is used for runtimeclass scheduling, not just pod overhead.

Author: Yeah... I was thinking about it. Making it stable creates false impression that PodOverhead is also stable. @sftim what would you suggest? RuntimeClass feature is stable, PodOverhead which is part of the same API is in beta

Author: This is exactly what I asked on the last SIG node meeting as I also lack context on how it is typically handled in k8s. So the outcome of the discussion was that this is fine. It is similar to the situation when the new feature being enabled via feature gate on the already stable API. So the text you suggested looks accurate

Author: ```suggestion
{{< feature-state for_k8s_version="v1.20" state="stable" >}}
```

> For [RuntimeClass](/docs/concepts/containers/runtime-class/) definitions which describe an overhead associated with running a pod,
> this admission controller will set the pod.Spec.Overhead field accordingly.

Reviewer 3: Is this accurate:
```suggestion
If you enable the `PodOverhead` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/), and define a RuntimeClass with [Pod overhead](/docs/concepts/scheduling-eviction/pod-overhead/) configured, this admission controller checks incoming
Pods. When enabled, this admission controller rejects any Pod create requests that have the overhead already set.
For Pods that have a  RuntimeClass is configured and selected in their `.spec`, this admission controller sets `.spec.overhead` in the Pod based on the value defined in the corresponding RuntimeClass.

{{< note >}}
The `.spec.overhead` field for Pod and the `.overhead` field for RuntimeClass are both in beta. If you do not enable the `PodOverhead` feature gate, all Pods are treated as if `.spec.overhead` is unset.
{{< /note >}}
```
